libraries/WiFi/src/WiFiMulti.cpp

```diff
@@ -118,17 +118,19 @@ uint8_t WiFiMulti::run(uint32_t connectTimeout)
                     status = WiFi.status();
                 }
 
-                IPAddress ip;
-                uint8_t * mac;
                 switch(status) {
                 case 3:
-                    ip = WiFi.localIP();
-                    mac = WiFi.BSSID();
+                    {
+#if ARDUHAL_LOG_LEVEL >= ARDUHAL_LOG_LEVEL_DEBUG
+                    IPAddress ip = WiFi.localIP();
```

Comment:
you can not declare new variables inside the case. Add another `#if` for their definitions above.

Reply:
the variable is not necessary at all even, the log_d line can be updated to just:
```
log_d("[WIFI] IP: %s", WiFi.localIP().toString().c_str());
```

Reply:
same for BSSID probably with WiFi.BSSIDstr()

Reply:
Ok I will do 
For my personnal knowledge - as I have set brace, why  it is not correct to declare new local variables inside the case ?